prov/shm/src/smr_progress.c

```diff
@@ -513,11 +525,17 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 						     ptr, cmd->msg.hdr.size);
 	}
 
-	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE)
+	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE) {
 		close(ipc_fd);
-
-	/* Truncation error takes precedence over close_handle error */
-	ret = ofi_hmem_close_handle(cmd->msg.data.ipc_info.iface, base);
+		/* Truncation error takes precedence over close_handle error */
+		ret = ofi_hmem_close_handle(cmd->msg.data.ipc_info.iface, base);
+	} else {
+		/* 
+		 * Decrement the use_cnt of the ipc mr_entry
+		 * so it can be cleaned properly 
+		 */
```

Comment:
Drop the comment -- "use_cnt" is some internal thing within the cache.  Search should simply be paired with delete.